crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/enums/BenefitDetails.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.justice.laa.crime.applications.adaptor.enums;
+
+import java.util.Arrays;
+
+public enum BenefitDetails {
+    CHILD("child", "CHILD_BEN"),
+    WORKING_OR_CHILD_TAX_CREDIT("working_or_child_tax_credit", "TAX_CRED"),
+    INCAPACITY("incapacity", "INCAP_BEN"),
+    INDUSTRIAL_INJURIES_DISABLEMENT("industrial_injuries_disablement", "INJ_BEN"),
+    OTHER("other", Constants.OTHER_BEN),
+    UNIVERSAL_CREDIT("universal_credit", Constants.OTHER_BEN),
+    JSA("jsa", Constants.OTHER_BEN);
+
+    private final String value;
+    private final String code;
+
+    private static class Constants {
+        private static final String OTHER_BEN ="OTHER_BEN";
+    }
+
+    BenefitDetails(String value, String code) {
+        this.value = value;
+        this.code = code;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static BenefitDetails findByValue(String value) {
+        return Arrays.stream(values()).filter(benefitDetails -> benefitDetails.getValue().equalsIgnoreCase(value)).findFirst().orElseThrow(IllegalArgumentException::new);
```

Comment:
Do we want to throw the exception here or at least have it in the method signature?

Reply:
There is no chance to throw an exception here as we are mapping values from the enum and we have valid data from the Crime Apply data store. there is no chance to get any other values from the CA datastore
As per the functional programming style, no need to add throw exception in the method signature

Reply:
I'm talking about this part: .orElseThrow(IllegalArgumentException::new)